docs/resources/mws_ncc_binding.md

```diff
@@ -0,0 +1,45 @@
+---
+subcategory: "Deployment"
+---
+# databricks_mws_ncc_binding Resource
+
+-> **Note** Initialize provider with `alias = "account"`, `host = "https://accounts.azuredatabricks.net"` and use `provider = databricks.account` for all `databricks_mws_*` resources.
+
+-> **Public Preview** This feature is available for AWS & Azure only, and is in [Public Preview](https://docs.databricks.com/release-notes/release-types.html) in AWS.
+
+Allows you to attach a [Network Connectivity Config](mws_network_connectivity_config) object to a [databricks_mws_workspaces](mws_workspaces.md) resource to create a [Databricks Workspace that leverages serverless network connectivity configs](https://learn.microsoft.com/en-us/azure/databricks/sql/admin/serverless-firewall).
+
+The NCC and workspace must be in the same region.
+
+## Example Usage
+
+```hcl
+variable "region" {}
+variable "databricks_workspace_id" {}
+variable "prefix" {}
+
+resource "databricks_mws_network_connectivity_config" "ncc" {
+  provider = databricks.account
+  name     = "Network Connectivity Config for ${var.prefix}"
+  region   = var.region
+}
+
+resource "databricks_mws_ncc_binding" "ncc_binding" {
+  provider                       = databricks.account
+  network_connectivity_config_id = databricks_mws_network_connectivity_config.ncc.id
+  workspace_id                   = var.databricks_workspace_id
+}
+```
+
+## Argument Reference
+
+The following arguments are available:
+
+* `network_connectivity_config_id` - Canonical unique identifier of Network Connectivity Config in Databricks Account.
```

Comment:
Changes force the creation of a new resource?

Reply:
@nkvuong  just to confirm, is this intentional for this resource because we don't set it as force_new for `mws_ncc_binding` but we do this for `resource_mws_ncc_private_endpoint_rule`
````
for _, p := range []string{"network_connectivity_config_id", "group_id", "resource_id"} {
			common.CustomizeSchemaPath(m, p).SetRequired().SetForceNew()
}
````

Reply:
@tanmay-db each workspace can only be bound to a single ncc, so this field is updatable